### PR TITLE
vimPlugins: Refactor coc-* plugins and fix coc-go

### PR DESCRIPTION
**Pull request by jbaum98**
###### Motivation for this change

I had an issue getting `coc-go` to work. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Result of `nixpkgs-review` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>25 packages built:</summary>
<br>- vimPlugins.Tagbar (vimPlugins.tagbar)
<br>- vimPlugins.clang_complete
<br>- vimPlugins.coc-explorer
<br>- vimPlugins.coc-go
<br>- vimPlugins.coc-markdownlint
<br>- vimPlugins.completion-nvim
<br>- vimPlugins.csv (vimPlugins.csv-vim)
<br>- vimPlugins.denite (vimPlugins.denite-nvim)
<br>- vimPlugins.deol-nvim
<br>- vimPlugins.nvim-lspconfig
<br>- vimPlugins.nvim-treesitter
<br>- vimPlugins.polyglot (vimPlugins.vim-polyglot)
<br>- vimPlugins.vim-clap
<br>- vimPlugins.vim-dirvish
<br>- vimPlugins.vim-floaterm
<br>- vimPlugins.vim-flutter
<br>- vimPlugins.vim-html-template-literals
<br>- vimPlugins.vim-lsp
<br>- vimPlugins.vim-monokai
<br>- vimPlugins.vim-sneak
<br>- vimPlugins.vim-test
<br>- vimPlugins.vim-vsnip
<br>- vimPlugins.vimspector
<br>- vimPlugins.vimtex
<br>- vimPlugins.zenburn
</details>

### CODE REVIEW

**Comment by jbaum98** [comment]
I realized this accomplishes the same thing as #82723, but also does the refactor and removes them from vim-plugin-names.

**Comment by jbaum98** [comment]
Yup I can do that.

**Review by jonringer — APPROVED** [review]
diff LGTM
gets rid of a bunch of boilerplate

```
https://github.com/NixOS/nixpkgs/pull/96993
25 packages built:
vimPlugins.clang_complete vimPlugins.coc-explorer vimPlugins.coc-go vimPlugins.coc-markdownlint vimPlugins.completion-nvim vimPlugins.csv-vim vimPlugins.denite-nvim vimPlugins.deol-nvim vimPlugins.nvim-lspconfig vimPlugins.nvim-treesitter vimPlugins.tagbar vimPlugins.vim-clap vimPlugins.vim-dirvish vimPlugins.vim-floaterm vimPlugins.vim-flutter vimPlugins.vim-html-template-literals vimPlugins.vim-lsp vimPlugins.vim-monokai vimPlugins.vim-polyglot vimPlugins.vim-sneak vimPlugins.vim-test vimPlugins.vim-vsnip vimPlugins.vimspector vimPlugins.vimtex vimPlugins.zenburn
```

still would like to hear from @timokau @evanjs

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -791,4 +595,47 @@ self: super: {
...
+      "coc-tsserver"
+      "coc-vetur"
+      "coc-vimtex"
+      "coc-wxml"
+      "coc-yaml"
+      "coc-yank"
+    ];
+    nodePackage2VimPackage = name: buildVimPluginFrom2Nix {
+      pname = name;
+      inherit (nodePackages.${name}) version;
```
Comment (timokau) [inline pkgs/misc/vim-plugins/overrides.nix:636]: Would it make sense to inherit `meta` (description and license) as well?

**Review by timokau — APPROVED** [review]
Generally looks good to me, though I'm not very familiar with how coc was handled in the past and I have never used it.

**Comment by gvolpe** [comment]
@jbaum98 Right now CoC plugins can be accessed via `pkgs.vimPlugins.coc-*`. Would it remain the same after these changes or how would it be?

File: pkgs/misc/vim-plugins/overrides.nix (hunk 1 above)
Comment (jbaum98) [reply pkgs/misc/vim-plugins/overrides.nix:636]: Oh that's a very good idea.

**Comment by jbaum98** [comment]
Yes, they'll still be available in `vimPlugins`. What this does is removes the non-compiled versions, since they were overriden anyway, and changes the way they are added to `vimPlugins` in `overrides.nix` to reduce duplication.

**Comment by jonringer** [comment]
@gvolpe Yea, I re-did my home-manager off this PR and it seems that the plugin names have stayed "the same"

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 96993 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>25 packages built:</summary>  <ul>    <li>vimPlugins.clang_complete</li>    <li>vimPlugins.coc-explorer</li>    <li>vimPlugins.coc-go</li>    <li>vimPlugins.coc-markdownlint</li>    <li>vimPlugins.completion-nvim</li>    <li>vimPlugins.csv-vim</li>    <li>vimPlugins.denite-nvim</li>    <li>vimPlugins.deol-nvim</li>    <li>vimPlugins.nvim-lspconfig</li>    <li>vimPlugins.nvim-treesitter</li>    <li>vimPlugins.tagbar</li>    <li>vimPlugins.vim-clap</li>    <li>vimPlugins.vim-dirvish</li>    <li>vimPlugins.vim-floaterm</li>    <li>vimPlugins.vim-flutter</li>    <li>vimPlugins.vim-html-template-literals</li>    <li>vimPlugins.vim-lsp</li>    <li>vimPlugins.vim-monokai</li>    <li>vimPlugins.vim-polyglot</li>    <li>vimPlugins.vim-sneak</li>    <li>vimPlugins.vim-test</li>    <li>vimPlugins.vim-vsnip</li>    <li>vimPlugins.vimspector</li>    <li>vimPlugins.vimtex</li>    <li>vimPlugins.zenburn</li>  </ul></details>

**Comment by jbaum98** [comment]
Wait! I wanted to inherit `meta` like timokau suggested.

**Comment by jonringer** [comment]
that shouldn't cause any rebuilds, :) just open a PR! xD

**Comment by jonringer** [comment]
sorry, I'm trying to do a "last call" of merges before branchoff

**Pull request merged**